dhis-2/dhis-api/src/main/java/org/hisp/dhis/util/DateUtils.java

```diff
@@ -672,6 +675,28 @@ public static Date parseDate(String dateString) {
     return safeParseDateTime(dateString, DATE_FORMATTER);
   }
 
+  /**
+   * Parses the given string into a Date using the supported date formats. Add time at the end of
+   * the day if no time was provided. Returns null if the string cannot be parsed.
+   *
+   * @param dateString the date string.
+   * @return a date.
+   */
+  public static Date parseDateEndOfTheDay(String dateString) {
+    if (StringUtils.isEmpty(dateString)) {
+      return null;
+    }
+
+    try {
+      Date date = safeParseDateTime(dateString, ONLY_DATE_FORMATTER);
+      org.joda.time.LocalDateTime localDateTime = org.joda.time.LocalDateTime.fromDateFields(date);
```

Comment:
Could we do this without relying on an external library?
Here's [an example](https://stackoverflow.com/questions/10308356/how-to-obtain-the-start-time-and-end-time-of-a-day) with java8

Reply:
Done